pkg/controller/elasticsearch/nodespec/statefulset.go

```diff
@@ -118,6 +116,29 @@ func BuildStatefulSet(
 	return sset, nil
 }
 
+func setVolumeClaimsControllerReference(
+	persistentVolumeClaims []corev1.PersistentVolumeClaim,
+	es v1beta1.Elasticsearch,
+	scheme *runtime.Scheme,
+) ([]corev1.PersistentVolumeClaim, error) {
+	// set the owner reference of all volume claims to the ES resource,
+	// so PVC get deleted automatically upon Elasticsearch resource deletion
+	claims := make([]corev1.PersistentVolumeClaim, 0, len(persistentVolumeClaims))
+	for _, claim := range persistentVolumeClaims {
+		if err := controllerutil.SetControllerReference(&es, &claim, scheme); err != nil {
+			return nil, err
+		}
+		// Set block owner deletion to false as the statefulset controller might not be able to do that
```

Comment:
> to do that

Maybe needs half a sentence more. IIUC the statefulset controller cannot handle an owner reference with `blockOwnerDeletion:true` if it cannot set finalizers on the owning resource